src/core/lib/transport/transport.cc

```diff
@@ -22,22 +22,23 @@
 
 #include <string.h>
 
+#include <memory>
 #include <new>
 
 #include "absl/status/status.h"
 
+#include <grpc/event_engine/event_engine.h>
+
+#include "src/core/lib/event_engine/default_event_engine.h"
 #include "src/core/lib/gpr/alloc.h"
 #include "src/core/lib/iomgr/exec_ctx.h"
-#include "src/core/lib/iomgr/executor.h"
-#include "src/core/lib/iomgr/iomgr.h"
 #include "src/core/lib/transport/transport_impl.h"
 
 grpc_core::DebugOnlyTraceFlag grpc_trace_stream_refcount(false,
                                                          "stream_refcount");
 
 void grpc_stream_destroy(grpc_stream_refcount* refcount) {
-  if (!grpc_iomgr_is_any_background_poller_thread() &&
```

Comment:
Could you clarify why we're removing the check on grpc_iomgr_is_any_background_poller_thread() ?

Reply:
it's an optimization that I think is hard to replicate generally going forward, and one that is so limited in scope (the execctx condition is very rarely triggered) that i think we're safe to lose it

Reply:
I see.